changelog/7469.improvement.rst

```diff
@@ -0,0 +1,23 @@
+It is now possible to construct a :class:`MonkeyPatch` object directly as ``pytest.MonkeyPatch()``,
+in cases when the :fixture:`monkeypatch` fixture cannot be used. Previously some users imported it
+from the private `_pytest.monkeypatch.MonkeyPatch` namespace.
+
+The types of builtin pytest fixtures are now exported so they may be used in type annotations of test functions.
+The newly-exported types are:
+
+- ``pytest.FixtureRequest`` for the :fixture:`request` fixture.
+- ``pytest.Cache`` for the :fixture:`cache` fixture.
+- ``pytest.CaptureFixture[str]`` for the :fixture:`capfd` and :fixture:`capsys` fixtures.
+- ``pytest.CaptureFixture[bytes]`` for the :fixture:`capfdbinary` and :fixture:`capsysbinary` fixtures.
+- ``pytest.LogCaptureFixture`` for the :fixture:`caplog` fixture.
+- ``pytest.Pytester`` for the :fixture:`pytester` fixture.
+- ``pytest.Testdir`` for the :fixture:`testdir` fixture.
+- ``pytest.TempdirFactory`` for the :fixture:`tmpdir_factory` fixture.
+- ``pytest.TempPathFactory`` for the :fixture:`tmp_path_factory` fixture.
+- ``pytest.MonkeyPatch`` for the :fixture:`monkeypatch` fixture.
+- ``pytest.WarningsRecorder`` for the :fixture:`recwarn` fixture.
+
+Constructing them is not supported (except for `MonkeyPatch`); they are only meant for use in type annotations.
+Doing so will emit a deprecation warning, and may become a hard-error in pytest 7.0.
+
+Subclassing them is also not supported. This is not currently enforced at runtime, but is detected by type-checkers such as mypy.
```

Comment:
it's quite clean to add a runtime deprecation warning, so might be worth doing in this PR:

```python
def __init_subclass__(cls, **kwargs):
    warnings.warn(f"subclassing {cls} is deprecated", PytestDeprecationWarning)
    super().__init_subclass__(**kwargs)
```

Reply:
Interesting, I'll open an issue to track this suggestion.